src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -359,15 +370,45 @@ public void Train(double[] values, ref double[][] results)
                 {
                     Array.Resize<double[]>(ref results, values.Length);
                 }
-                SpectralResidual(values, results, _threshold);
+
+                bool isTemporal = true;
+                double[] seriesToDetect = values.ToArray();
+
+                if (_period > 0)
+                {
+                    StlConfiguration config = new StlConfiguration(_period);
+                    InnerStl stl = new InnerStl(values, config, isTemporal);
+                    bool success = stl.Decomposition();
+
+                    if (success)
+                    {
+                        seriesToDetect = stl.Residual.ToArray();
```

Comment:
>ToArray [](start = 54, length = 7)

Please try not to allocate new arrays on each call.

Reply:
fixed.

---
In reply to: [435226210](https://github.com/dotnet/machinelearning/pull/5202#discussion_r435226210) [](ancestors = 435226210)